src/go/rpk/pkg/cli/registry/mode/set.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package mode
+
+import (
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/schemaregistry"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/sr"
+)
+
+var supportedModes = []string{"READONLY", "READWRITE"}
```

Comment:
is this case insensitive?

Reply:
In practice: no, you can pass `readonly`, `read-only`, and other combinations... but we only document one so it doesn't become overcomplicated to explain.

Reply:
We can add in the help text something like "accepted values: readonly, readwrite (case insensitive)" 

I know we have this for other rpk commands like rpk security user update

Reply:
Done :+1: